android/src/main/java/com/reactnativestripesdk/PaymentSheetAppearance.kt

```diff
@@ -0,0 +1,136 @@
+package com.reactnativestripesdk
+
+import android.graphics.Color
+import android.os.Bundle
+import com.stripe.android.paymentsheet.PaymentSheet
+
+fun PaymentSheetFragment.buildPaymentSheetAppearance(userParams: Bundle?): PaymentSheet.Appearance {
+  val colorParams = userParams?.getBundle("colors")
+  val lightColorParams = colorParams?.getBundle("light") ?: colorParams
+  val darkColorParams = colorParams?.getBundle("dark") ?: colorParams
+
+  return PaymentSheet.Appearance(
+    typography = buildTypography(userParams?.getBundle("font")),
+    colorsLight = buildColors(lightColorParams, PaymentSheet.Colors.defaultLight),
+    colorsDark = buildColors(darkColorParams, PaymentSheet.Colors.defaultDark),
+    shapes = buildShapes(userParams?.getBundle("shapes")),
+    primaryButton = buildPrimaryButton(userParams?.getBundle("primaryButton"))
+  )
+}
+
+private fun PaymentSheetFragment.buildTypography(fontParams: Bundle?): PaymentSheet.Typography {
+  return PaymentSheet.Typography.default.copy(
+    sizeScaleFactor = getFloatOr(fontParams, "scale", PaymentSheet.Typography.default.sizeScaleFactor),
+    fontResId = getFontResId(fontParams, "family", PaymentSheet.Typography.default.fontResId)
+  )
+}
+
+private fun colorFromHexOrDefault(hexString: String?, default: Int): Int {
+  return hexString?.trim()?.replace("#","")?.let {
+    Color.parseColor("#$it")
+  } ?: run {
+    default
+  }
+}
+
+private fun buildColors(colorParams: Bundle?, default: PaymentSheet.Colors): PaymentSheet.Colors {
+  if (colorParams == null) {
+    return default
+  }
+
+  return default.copy(
+    primary = colorFromHexOrDefault(colorParams.getString("primary"), default.primary),
```

Comment:
up to you, but I think it might be good to define all of these strings in a companion object. We use them in a few place and it might prevent typos.

Reply:
yeah good idea!

Reply:
addressed for both ios and android in https://github.com/stripe/stripe-react-native/pull/940/commits/fb680dd5c1a256ce87617ac9ece4755743e5f6f6